### PR TITLE
vim-full: rename from vim_configurable

### PR DESCRIPTION
**Pull request by ncfavier**
Supersedes https://github.com/NixOS/nixpkgs/pull/145351.

The word "configurable" has [proven](https://github.com/NixOS/nixpkgs/pull/201380#discussion_r1038241962) to be confusing because it suggests user configuration, not build configuration. I think this is enough to warrant a rename.

Makes `vim_configurable` and `vimHugeX` aliases to `vim-full`.

### CODE REVIEW

**Review by Ma27 — APPROVED** [review]

**Comment by ncfavier** [comment]
@ofborg eval

File: nixos/doc/manual/from_md/release-notes/rl-2305.section.xml
Comment (teto) [inline nixos/doc/manual/from_md/release-notes/rl-2305.section.xml:139]: maybe append `i.e, .vimrc ` ?

**Comment by teto** [comment]
"vim-full" is much better than "vim_configurable". Is there a plan to retire  "vim_configurable" ?

**Review by millerjason — APPROVED** [review]

**Comment by ncfavier** [comment]
> Is there a plan to retire "vim_configurable" ?

I don't know what is usually done for deprecated aliases. I don't think it's an emergency to retire it, since it won't show up in search any more. Maybe in a few releases when everyone has forgotten about it.

**Review by teto — APPROVED** [review]
I'll let the vim maintainers merge.

File: doc/languages-frameworks/vim.section.md
Comment (SuperSandro2000) [inline doc/languages-frameworks/vim.section.md:20]:
```suggestion
::: {.note}
`vim_configurable` is a deprecated alias for `vim-full` and refers to the fact that its
build-time features are configurable. It has nothing to do with user configuration,
and both the `vim` and `vim-full` packages can be customized as explained in the next section.
:::
```

File: doc/languages-frameworks/vim.section.md
Comment (SuperSandro2000) [inline doc/languages-frameworks/vim.section.md:14]:
```suggestion
Right now two Vim packages are available: `vim` which has most features that require extra dependencies disabled and `vim-full` that has them enabled.
```

File: pkgs/applications/editors/neovim/tests/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/editors/neovim/tests/default.nix:2]:
```suggestion
{ vimUtils, writeText, neovim, vimPlugins
```
If it is unused we might as well remove it

**Comment by ncfavier** [comment]
We probably shouldn't block this on the vim maintainers.

**Review by equirosa — APPROVED** [review]
LGTM

**Review by millerjason — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> We probably shouldn't block this on the vim maintainers.

We don't have strong maintainer requirements so really anyone can merge this.

**Comment by dasJ** [comment]
> We probably shouldn't block this on the vim maintainers.

I agree with this as a vim maintainer since I don't have time for nixpkgs all the time.
Changes like this are an improvement to the overall consistency of nixpkgs and always highly appreciated.

**Pull request merged**
